Add Continuous integration Workflow and enable project to be compiled with JDK 11

When building with JDK 11 the following error occurs:

```
[INFO]
[INFO] --- maven-checkstyle-plugin:2.13:checkstyle (check-style) @ jboss-common-beans ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.052 s
[INFO] Finished at: 2023-04-12T15:27:44+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.13:checkstyle (check-style) on project jboss-common-beans: Execution check-style of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.13:checkstyle failed: Plugin org.apache.maven.plugins:maven-checkstyle-plugin:2.13 or one of its dependencies could not be resolved: Could not find artifact com.sun:tools:jar:1.7.0 at specified path /Library/Java/JavaVirtualMachines/jdk-11.0.13.jdk/Contents/Home/../lib/tools.jar -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

In addition, there is no CI enabled yet

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,0 +1,30 @@
+name: JBoss Common Beans CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build:
+    name: ${{ matrix.os }}-jdk${{ matrix.java }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        java: ['8', '11']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+          cache: maven
+      - name: Build and Test on ${{ matrix.java }}
+        run: mvn clean install

--- a/pom.xml
+++ b/pom.xml
@@ -12,7 +12,7 @@
     <parent>
         <groupId>org.jboss</groupId>
         <artifactId>jboss-parent</artifactId>
-        <version>17</version>
+        <version>39</version>
     </parent>
 
     <groupId>org.jboss.common</groupId>
@@ -45,8 +45,11 @@
             For example: <version.org.jboss.as.console> -->
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <version.compiler.plugin>3.1</version.compiler.plugin>
-        <version.org.wildfly.checkstyle-config>1.0.3.Final</version.org.wildfly.checkstyle-config>
+        <version.org.wildfly.checkstyle-config>1.0.8.Final</version.org.wildfly.checkstyle-config>
         <version.junit>4.12</version.junit>
+
+        <version.surefire.plugin>2.18.1</version.surefire.plugin>
+        <version.failsafe.plugin>${version.surefire.plugin}</version.failsafe.plugin>
 
         <!-- Checkstyle configuration -->
         <linkXRef>false</linkXRef>

